### ISSUE TITLE
Sieve filters: error message not clear

### ISSUE THREAD

**Issue by minami-o** [RainLoop/rainloop-webmail#570]
Hi there,
First off, thanks for this great tool you built and shared!
I am a heavy user of sieve filters, and realized that, when using a manual script to set up filters, the error message in case of syntax error is just "Can't save filters", which I think is a bit short...
The syntax check works well, but the error message is just not appropriate.
The logs gave me a 351 error code for this matter, but I could find no documentation on sieve error codes, so I can't guarantee it's the right/only one for syntax errors.
Could you please try and make this error message a bit clearer?
Cheers!
> AJAX[DATA]: {"Action":"FiltersSave","Result":false,"ErrorCode":351,"ErrorMessage":"","ErrorMessageAdditional":"","Time":93}

**Comment by RainLoop** [comment]
351 is a internal error (RainLoop). it means "Can't save filters".
I need to see a sieve error code.

Can you give me (support@rainloop.net) the full log file.

**Comment by minami-o** [comment]
Thanks for your quick reply!
The error message from ManageSieve seems to be "MailSo-Sieve-Exceptions-NegativeResponseException", as you can see in the file I just sent you.
I'm afraid *this* error is a bit generic, but let's hope you can still do something with it :)

**Comment by RainLoop** [comment]
Well, your server says:
line 171: error: expected end of command ';' or the beginning of a compound block '{', but found ','

Something wrong with your sieve script on 171 line.

**Comment by minami-o** [comment]
Mmm I think there is a misunderstanding here: I found the syntax error before posting, I corrected it and my script is now OK.
My point was that, at the time my script was wrong, RL prevented me from saving the filter - which is good - but the error message it gave me was just "Can't save filters", which I find is not very accurate.
I would have prefered a message like "Syntax error" or, even better "Syntax error on line 171".
Even the error message you isolated would be good, actually.
> line 171: error: expected end of command ';' or the beginning of a compound block '{', but found ','

Is it any clearer now?
